tests/tests/market_fee_sharing_tests.cpp

```diff
@@ -576,15 +535,6 @@ BOOST_AUTO_TEST_CASE(create_asset_via_proposal_test)
       prop.expiration_time =  db.head_block_time() + fc::days(1);
       prop.fee = asset( proposal_create_fees.fee + proposal_create_fees.price_per_kbyte );
 
-      {
-         signed_transaction tx;
-         tx.operations.push_back( prop );
-         db.current_fee_schedule().set_fee( tx.operations.back() );
-         set_expiration( db, tx );
-         sign( tx, issuer_private_key );
-         GRAPHENE_CHECK_THROW(PUSH_TX( db, tx ), fc::exception);
-      }
-
       generate_blocks_past_hf1268();
```

Comment:
I thought we had removed all hf check code about hf1268, why leave it here then?

Reply:
Unfortunately, some must remain. See db_market.cpp. pay_market_fees() needs an extra parameter after the hf.

Reply:
I don't think the parameter is necessary. Before the hard fork, the fee sharing field is always unset, so theoretically it will work without the parameter. A minor code refactory would be required.

Reply:
pay_market_fees is called by

- fill_settle_order, which always calls the old version of the method
- fill_limit_order, which calls the correct version of the method, depending on head_block_time().

Both versions of the method check to assure that there are issuer_fees to be paid before going any further. The new version includes a section to determine if rewards are allowed. If they are not allowed, it does what the old version does. My suggestion for refactoring would be to add logic to is_rewards_allowed to check that seller != graphene::chain::account_object(), and modify both callers to always call the new method.

As for performance, this will avoid the head_block_time() call, the comparison with HARDFORK_1268_TIME, but add a comparison within pay_market_fees().

1. Does my suggestion for refactoring above make sense?
1. Is there a better way of doing the seller comparison ( i.e. avoid the temporary object creation) while maintaining readability?